src/core/tranche/manager.sol

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2019 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "ds-note/note.sol";
+import "../lightswitch.sol";
+
+contract PileLike {
+    function want() public returns (int);
+}
+
+contract OperatorLike {}
+
+contract DistributorLike {
+    function balance() public;
+    function repayTranches(uint) public;
+}
+
+// TrancheManager
+// Keeps track of the tranches. Manages the interfacing between the tranche side and borrower side of the contracts.
+contract TrancheManager is DSNote {
+
+    // --- Auth ---
+    mapping (address => uint) public wards;
+    function rely(address usr) public auth note { wards[usr] = 1; }
+    function deny(address usr) public auth note { wards[usr] = 0; }
+    modifier auth { require(wards[msg.sender] == 1); _; }
+
+    // --- Data ---
+    DistributorLike public distributor;
+    PileLike public pile;
+
+    // --- Tranches ---
+    OperatorLike public senior;
```

Comment:
can the junior & senior vars not just be of type address

Reply:
yes, we could.  we might rename it to `seniorOperator` to highlight it is the operator contract address.